src/main/java/com/github/miosz/carservice/controller/CarController.java

```diff
@@ -0,0 +1,69 @@
+package com.github.miosz.carservice.controller;
+
+import com.github.miosz.carservice.model.Car;
+import com.github.miosz.carservice.service.CarService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@Controller
+public class CarController {
+
+    final CarService carService;
+
+    public CarController(CarService carService) {
+        this.carService = carService;
+    }
+
+    @GetMapping("/cars")
+    public String getCarsToRepair(Model model) {
+        model.addAttribute("cars", carService.getCarsToFix());
+        return "cars";
+    }
+
+    @GetMapping("/cars/add")
+    public String getAddCars(Model model) {
+        model.addAttribute("car", new Car());
+        return "cars-add-form";
+    }
+
+    @GetMapping("/cars/fix")
+    public String getCarsToFix(Model model) {
+        model.addAttribute("cars", carService.getCarsToFix());
+        return "cars-fix";
+    }
+
+    @PostMapping("/cars/fix/{registrationNumber}")
+    public String postFixCar(@PathVariable("registrationNumber") String registrationNumber, Model model) {
+        Car car = carService.getCarByRegistrationNumber(registrationNumber);
+        carService.removeFixedCar(car);
+        carService.saveCarsToFixToFile();
+        carService.fixCar(car);
+        carService.setFixedDate(car);
+        carService.getFixedCars().add(car);
+        carService.saveFixedCarToFile(carService.getFixedCars());
+        model.addAttribute("cars", carService.getCarsToFix());
+        return "redirect:/cars/fix";
```

Comment:
Chyba nie musimy wywoływać `getCarsToFix` skoro po redirect do endpointu `/cars/fix` robimy to samo 😉